ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Network/NodeToNode.hs

```diff
@@ -577,9 +579,15 @@ mkApps kernel Tracers {..} mkCodecs ByteLimits {..} genChainSyncTimeout ReportPe
                   (timeLimitsChainSync chainSyncTimeout)
                   channel
                   $ chainSyncClientPeerPipelined
-                  $ hChainSyncClient them isBigLedgerPeer version controlMessageSTM
-                      (TraceLabelPeer them `contramap` reportHeader)
-                      varCandidate
+                  $ hChainSyncClient
+                      them
+                      isBigLedgerPeer
+                      CsClient.DynamicEnv {
+                          CsClient.version
```

Comment:
I find it confusing that a qualified field inherits the unqualified value of the same unqualified name.

Reply:
Yeahs, it's a weird corner case of `-XNamedFieldPuns`.